File: src/Klc.LogicRoute.Domain/Entities/ProofOfDelivery.cs
using Klc.LogicRoute.Domain.Common;

namespace Klc.LogicRoute.Domain.Entities;

public class ProofOfDelivery : BaseEntity
{
    public Guid ShipmentId { get; set; }
    public string? PhotoPath { get; set; }
    public string? SignaturePath { get; set; }
    public string? RecipientName { get; set; }
    public string? Notes { get; set; }
    public double? Lat { get; set; }
    public double? Lng { get; set; }
    public DateTime CapturedAt { get; set; } = DateTime.UtcNow;
}

File: mobile/composeApp/src/commonMain/kotlin/com/klc/logicroute/data/model/ProofOfDelivery.kt
package com.klc.logicroute.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ProofOfDelivery(
    @SerialName("shipmentId")
    val shipmentId: String,
    @SerialName("recipientName")
    val recipientName: String,
    @SerialName("notes")
    val notes: String = "",
    @SerialName("timestamp")
    val timestamp: String = ""
)
